stable/redis/values.yaml

```diff
@@ -61,14 +61,29 @@ networkPolicy:
   ##
   # allowExternal: true
 
+# Disable redis auth for master and slave by setting this to false
+usePassword: true
+
 ## Redis password (both master and slave)
-## Defaults to a random 10-character alphanumeric string if not set and usePassword is true
+## Defaults to a random 10-character alphanumeric string - not used if `usePassword` is set to false
 ## ref: https://github.com/bitnami/bitnami-docker-redis#setting-the-server-password-on-first-run
 ##
-usePassword: true
 password:
-## Use existing secret (ignores previous password)
-# existingSecret:
+
+# Define where redis should get its secrets from.
```

Comment:
What if this just pointed to a Secret name, but expected the same keys that it would create for you?

Reply:
The reason for introducing the `secretKeyRefs` concept is, to enable the user to gain full control of the lifetime and source of the secret.

We have some cases where we have an application which requires multiple redis instances; each instance requires it's own password, all passwords are coming from one secret - so this won't be possible with fixing the key-name.

If the default values allow the user to use the chart out-of-the-box, why not allowing the user to define the wanted key as well? 🙂

Reply:
IMO from a UX POV it's easier to provide a `secretName=my-existing-redis-secret` than providing the whole secretKeyRef block. I'm curious why you'd use a single secret with multiple passwords, that seems hard to manage and creating more secrets is fairly cheap in kubernetes.

Reply:
We are using one secret with multiple passwords to have one place where to look at for the applications secrets. So all secrets for a application release (helm release) stay together and the application controls the secrets for its required services as well; that makes debugging easier because the operator only needs to look at one secret instead of having to resolve secret references first.

Not specifying the secret key, requires the author of the external secret to specify the key in their secret resource.
But as I'm thinking through this; having a freeform secret key name for redis passwords results potentially in a zoo of redis password secrets-keys - but I would see this as a responsibility of the secret authors, in exchange of more power/control.

How would you handle those scenarios?

What would be the benefit of the reference, if I don't allow to change the key as well?

Reply:
I don't aim for a redis-chart specific solution here; but would love to come up with a more common way doing this.

Reply:
Okay, I see your point. You're right in keeping this as simple as it can!

So I will open an issue regarding this topic to invite more maintainers and may be we can have a more generic way. Currently it's hard to deal with all those approaches from all the different charts.

Despite the secret password topic; this PR is more than just referencing the password from a secret - it allows users to reference any ENV var from any secret they want the information from. (this is the original idea behind introducing a generic `secretKeyRefs`; dropping `existingSecret` is only done, because it could be achieved with the `secretKeyRefs` approach as well)
What do you think about that?

PS.: I really appreciate that you take the time discussing this PR that deeply! Thank you!

Reply:
> Despite the secret password topic; this PR is more than just referencing the password from a secret - it allows users to reference any ENV var from any secret they want the information from. (this is the original idea behind introducing a generic secretKeyRefs; dropping existingSecret is only done, because it could be achieved with the secretKeyRefs approach as well)
What do you think about that?

If you mean charts should allow you to add generic fields (like env vars) to templates, it's difficult because it becomes a slippery slope. The more Kubernetes YAML we move into values.yaml, the harder it is to take advantage of templating within Helm and closer you get to... well just writing Kubernetes templates :P.

If you want to do that, you might want to see if something like a PodPreset (https://kubernetes.io/docs/tasks/inject-data-application/podpreset/) does what you need instead. That's useful if you want to inject certain env vars across pods in a namespace.

There is a tradeoff here between customisability and a simple values.yaml API, and at some point it may be easier to maintain a fork of a chart to do what you need.

> PS.: I really appreciate that you take the time discussing this PR that deeply! Thank you!

absolutely, happy to help!